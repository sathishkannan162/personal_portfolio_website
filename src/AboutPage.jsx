import { Heading, Text, Image, Box, Link } from '@chakra-ui/react';
import PageContainer from './PageContainer';
import myPhoto from './assets/my_photo_about.jpg';
import StandardLink from './StandardLink';

export default function AboutPage() {
  return (
    <PageContainer>
      <Heading>About Me</Heading>
      <br />
      <Box
        maxW="400px"
        maxH="400px"
        float={{ base: 'none', md: 'left' }}
        pr={50}
      >
        <Image src={myPhoto} borderRadius={9} />
      </Box>
      <Text fontSize={'lg'}>
        I am Sathish Kannan. I recently graduated from ðŸŽ“IIT Bombay. I am a self
        taughtðŸ¤¸ full stack developer.
        <br />
        <br />
        Currently, I am building products that integrates with ðŸ¤–AI.
        <br />
        <br />
        Checkout my personal blog:{' '}
        <StandardLink href="https://sathishkannan162.github.io">
          {' '}
          https://sathishkannan162.github.io
        </StandardLink>
        <br />
        <br />
        Checkout my{' '}
        <StandardLink href="https://www.linkedin.com/in/sathish-kannan-a48a39166/">
          LinkedIn
        </StandardLink>
        ,{' '}
        <StandardLink href="https://twitter.com/Sathish06030039">
          Twitter
        </StandardLink>
        ,{' '}
        <StandardLink href="https://www.instagram.com/satishkannan162/">
          Instagram
        </StandardLink>{' '}
        and{' '}
        <StandardLink href="https://www.facebook.com/sathishkannan.m.3">
          Facebook
        </StandardLink>
        .
      </Text>
    </PageContainer>
  );
}
