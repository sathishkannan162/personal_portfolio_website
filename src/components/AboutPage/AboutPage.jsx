import { Heading, Text, Image, Box } from '@chakra-ui/react';
import PageContainer from '../CommonComponents/PageContainer';
import StandardLink from '../CommonComponents/StandardLink';

export default function AboutPage() {
  const myPhoto =
    'https://projectimages.netlify.app/my_photos/my_photo_about.jpg';
  return (
    <PageContainer data-test="about-page">
      <Heading>About Me</Heading>
      <br />
      <Box
        w={{ base: '300px', md: '400px' }}
        h={{ base: '300px', md: '400px' }}
        mx="auto"
        float={{ base: 'none', md: 'left' }}
        pr={{ base: 0, md: 50 }}
        pl={{ base: 0, md: 0 }}
      >
        <Image
          w={{ base: '300px', md: '350px' }}
          h={{ base: '300px', md: '350px' }}
          mx="auto"
          src={myPhoto}
          borderRadius={9}
        />
      </Box>
      <Box pt={{ base: 5, md: 0 }}>
        <Text fontSize={'lg'}>
          I am Sathish Kannan. I recently graduated from ðŸŽ“IIT Bombay. I am a
          self taughtðŸ¤¸ full stack developer.
          <br />
          <br />I have good technical proficienty in{' '}
          <strong>
            NextJs, React, Express, Nodejs and Data visualisation with D3.js{' '}
          </strong>
          <br />
          <br />
          Currently, I am building products that integrates with ðŸ¤–AI.
          <br />
          <br />
          Checkout my personal blog:{' '}
          <StandardLink href="https://sathishkannan162.github.io">
            {' '}
            https://sathishkannan162.github.io
          </StandardLink>
          <br />
          <br />
          Checkout my{' '}
          <StandardLink href="https://www.linkedin.com/in/sathish-kannan-a48a39166/">
            LinkedIn
          </StandardLink>
          ,{' '}
          <StandardLink href="https://twitter.com/Sathish06030039">
            Twitter
          </StandardLink>
          ,{' '}
          <StandardLink href="https://www.instagram.com/satishkannan162/">
            Instagram
          </StandardLink>{' '}
          and{' '}
          <StandardLink href="https://www.facebook.com/sathishkannan.m.3">
            Facebook
          </StandardLink>
          .
        </Text>
      </Box>
    </PageContainer>
  );
}
