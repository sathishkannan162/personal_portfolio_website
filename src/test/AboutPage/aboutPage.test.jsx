// eslint-disable-next-line no-unused-vars
import React from 'react';
import { render, screen } from '../test-utils'
// import { render, screen } from '@testing-library/react'
import AboutPage from '../../components/AboutPage/AboutPage'
import {ChakraProvider} from '@chakra-ui/react'

describe('AboutPage component', () => {
  it('renders heading', () => {
    render(<ChakraProvider><AboutPage /></ChakraProvider>)
    const headingElement = screen.getByTestId('about-page')
    expect(headingElement).toBeInTheDocument();
  });

  it('renders image with correct source', () => {
    const { getByRole } = render(<AboutPage />);
    const imageElement = getByRole('img');
    expect(imageElement).toHaveAttribute(
      'src',
      'https://projectimages.netlify.app/my_photos/my_photo_about.jpg'
    );
  });

  it('renders about page', () => {
    render(<AboutPage />);
    const textElement = screen.getByText('I am Sathish Kannan. I recently graduated from ðŸŽ“IIT Bombay. I am a self taughtðŸ¤¸ full stack developer.', { exact: false });
    expect(textElement).toBeInTheDocument();
  });

  it('renders personal blog link', () => {
    const { getByRole } = render(<AboutPage />);
    const blogLinkElement = getByRole('link', { name: 'https://sathishkannan162.github.io' });
    expect(blogLinkElement).toBeInTheDocument();
    expect(blogLinkElement).toHaveAttribute('href', 'https://sathishkannan162.github.io');
  });

  it('renders LinkedIn link', () => {
    const { getByRole } = render(<AboutPage />);
    const linkedInLinkElement = getByRole('link', { name: 'LinkedIn' });
    expect(linkedInLinkElement).toBeInTheDocument();
    expect(linkedInLinkElement).toHaveAttribute(
      'href',
      'https://www.linkedin.com/in/sathish-kannan-a48a39166/'
    );
  });
  it('renders Twitter link link', () => {
    const { getByRole } = render(<AboutPage />);
    const linkedInLinkElement = getByRole('link', { name: 'Twitter' });
    expect(linkedInLinkElement).toBeInTheDocument();
    expect(linkedInLinkElement).toHaveAttribute(
      'href',
      "https://twitter.com/Sathish06030039"
    );
  });
  it('renders Instagram link', () => {
  const { getByRole } = render(<AboutPage />);
  const instagramLinkElement = getByRole('link', { name: 'Instagram' });
  expect(instagramLinkElement).toBeInTheDocument();
  expect(instagramLinkElement).toHaveAttribute(
    'href',
    'https://www.instagram.com/satishkannan162/'
  );
});

it('renders Facebook link', () => {
  const { getByRole } = render(<AboutPage />);
  const facebookLinkElement = getByRole('link', { name: 'Facebook' });
  expect(facebookLinkElement).toBeInTheDocument();
  expect(facebookLinkElement).toHaveAttribute(
    'href',
    'https://www.facebook.com/sathishkannan.m.3'
  );
});
    it('captures and compares component snapshot', () => {
        const { asFragment } = render(<AboutPage />);
        expect(asFragment()).toMatchSnapshot();
    });
});
