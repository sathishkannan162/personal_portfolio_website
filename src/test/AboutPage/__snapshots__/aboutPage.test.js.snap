// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage component captures and compares component snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1qqblot"
    data-test="about-page"
  >
    <h2
      class="chakra-heading css-1dklj6k"
    >
      About Me
    </h2>
    <br />
    <div
      class="css-8g89q5"
    >
      <img
        class="chakra-image css-1kr5mmb"
        src="https://projectimages.netlify.app/my_photos/my_photo_about.jpg"
      />
    </div>
    <div
      class="css-gk7psq"
    >
      <p
        class="chakra-text css-i3jkqk"
      >
        I am Sathish Kannan. I recently graduated from ðŸŽ“IIT Bombay. I am a self taughtðŸ¤¸ full stack developer.
        <br />
        <br />
        I have good technical proficienty in 
        <strong>
          NextJs, React, Express, Nodejs and Data visualisation with D3.js 
        </strong>
        <br />
        <br />
        Currently, I am building products that integrates with ðŸ¤–AI.
        <br />
        <br />
        Checkout my personal blog: 
        <a
          class="chakra-link css-1vcqg6m"
          href="https://sathishkannan162.github.io"
          target="_blank"
        >
          <span
            class="chakra-text css-eq1gvl"
          >
             https://sathishkannan162.github.io
          </span>
        </a>
        <br />
        <br />
        Checkout my 
        <a
          class="chakra-link css-1vcqg6m"
          href="https://www.linkedin.com/in/sathish-kannan-a48a39166/"
          target="_blank"
        >
          <span
            class="chakra-text css-eq1gvl"
          >
            LinkedIn
          </span>
        </a>
        , 
        <a
          class="chakra-link css-1vcqg6m"
          href="https://twitter.com/Sathish06030039"
          target="_blank"
        >
          <span
            class="chakra-text css-eq1gvl"
          >
            Twitter
          </span>
        </a>
        , 
        <a
          class="chakra-link css-1vcqg6m"
          href="https://www.instagram.com/satishkannan162/"
          target="_blank"
        >
          <span
            class="chakra-text css-eq1gvl"
          >
            Instagram
          </span>
        </a>
         and 
        <a
          class="chakra-link css-1vcqg6m"
          href="https://www.facebook.com/sathishkannan.m.3"
          target="_blank"
        >
          <span
            class="chakra-text css-eq1gvl"
          >
            Facebook
          </span>
        </a>
        .
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
